rpc/core/blocks.go

```diff
@@ -141,6 +141,14 @@ func DataCommitment(ctx *rpctypes.Context, query string) (*ctypes.ResultDataComm
 		return nil, err
 	}
 
+	// limit of queried blocks
+	limit := 1000
+	if len(results) > limit {
+		return nil, fmt.Errorf("the query exceeds the limit of allowed blocks %d", limit)
+	} else if len(results) == 0 {
+		return nil, fmt.Errorf("cannot create the data commitments for an empty set of blocks")
+	}
```

Comment:
1) let's move this limit as a constant, preferably in the consts package

2) This does help in that we don't have to sort the blocks, which is awesome, but it doesn't prevent loading all the blocks in the first place, which may be the most resource expensive portion

Ideally we would check before attempting to get all the blocks, however, I do think it is still worth keeping, as it will at least serve as a warning to validators that something could be wrong with their query inputs. Regardless, we should be sure to include a check client side, which will catch this before the request is even sent. It can of course use the same constant listed above

Reply:
1. Where is the constants package ? I couldn't find it, that's why I created it there.
2. It is not loading the blocks. My bad, the function name was a bit misleading, I just changed it. But, up to that point, we still didn't query for them, we just knew which list of heights we should query for. (I just pushed a new commit fixing that)

Reply:
nice! 

and here https://github.com/SweeXordious/celestia-core/blob/dataCommitment_rpc/pkg/consts/consts.go

Reply:
Done. And added tests for the limit